catalog/to-consul/consul_node_services_client_ent_test.go

```diff
@@ -0,0 +1,343 @@
+					Address: "127.0.0.1",
+					Service: &api.AgentService{
+						ID:        "svc-id",
+						Service:   "svc",
+						Tags:      []string{"k8s"},
+						Namespace: "ns",
+					},
+				},
+			},
+			Exp: []ConsulService{
+				{
+					Namespace: "default",
+					Name:      "svc",
+				},
+				{
+					Namespace: "ns",
+					Name:      "svc",
+				},
+			},
+		},
+		"multiple services across multiple namespaces": {
+			ConsulServices: []api.CatalogRegistration{
+				{
+					Node:    "k8s-sync",
+					Address: "127.0.0.1",
+					Service: &api.AgentService{
+						ID:      "svc-id1",
+						Service: "svc",
+						Tags:    []string{"k8s"},
+					},
+				},
+				{
+					Node:    "k8s-sync",
+					Address: "127.0.0.1",
+					Service: &api.AgentService{
+						ID:      "svc-id2",
+						Service: "svc",
+						Tags:    []string{"k8s"},
+					},
+				},
+				{
+					Node:    "k8s-sync",
+					Address: "127.0.0.1",
+					Service: &api.AgentService{
+						ID:        "svc-id1",
+						Service:   "svc",
+						Tags:      []string{"k8s"},
+						Namespace: "ns",
+					},
+				},
+				{
+					Node:    "k8s-sync",
+					Address: "127.0.0.1",
+					Service: &api.AgentService{
+						ID:        "svc-id2",
+						Service:   "svc",
+						Tags:      []string{"k8s"},
+						Namespace: "ns",
+					},
+				},
+				{
+					Node:    "k8s-sync",
+					Address: "127.0.0.1",
+					Service: &api.AgentService{
+						ID:      "svc2-id1",
+						Service: "svc2",
+						Tags:    []string{"k8s"},
+					},
+				},
+				{
+					Node:    "k8s-sync",
+					Address: "127.0.0.1",
+					Service: &api.AgentService{
+						ID:      "svc2-id2",
+						Service: "svc2",
+						Tags:    []string{"k8s"},
+					},
+				},
+				{
+					Node:    "k8s-sync",
+					Address: "127.0.0.1",
+					Service: &api.AgentService{
+						ID:        "svc2-id1",
+						Service:   "svc2",
+						Tags:      []string{"k8s"},
+						Namespace: "ns",
+					},
+				},
+				{
+					Node:    "k8s-sync",
+					Address: "127.0.0.1",
+					Service: &api.AgentService{
+						ID:        "svc2-id2",
+						Service:   "svc2",
+						Tags:      []string{"k8s"},
+						Namespace: "ns",
+					},
+				},
+			},
+			Exp: []ConsulService{
+				{
+					Namespace: "default",
+					Name:      "svc",
+				},
+				{
+					Namespace: "default",
+					Name:      "svc2",
+				},
+				{
+					Namespace: "ns",
+					Name:      "svc",
+				},
+				{
+					Namespace: "ns",
+					Name:      "svc2",
+				},
+			},
+		},
+	}
+
+	for name, c := range cases {
+		t.Run(name, func(tt *testing.T) {
+			require := require.New(tt)
+			svr, err := testutil.NewTestServerT(tt)
+			require.NoError(err)
+			defer svr.Stop()
+
+			consulClient, err := api.NewClient(&api.Config{
+				Address: svr.HTTPAddr,
+			})
+			require.NoError(err)
+			for _, registration := range c.ConsulServices {
+				if registration.Service.Namespace != "" && registration.Service.Namespace != "default" {
+					_, _, err = consulClient.Namespaces().Create(&api.Namespace{
```

Comment:
For tests with multiple services in a non-default namespace, does this create that namespace multiple times?

Reply:
It makes the API call multiple times but no error is returned if the ns already exists